src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -63,6 +64,69 @@ public void Kill()
             }
         }
 
+        /// <summary>Stops the associated process immediately.</summary>
+        private void Stop()
+        {
+            EnsureState(State.HaveNonExitedId);
+            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGSTOP) != 0)
+            {
+                throw new Win32Exception(); // to match behavior of Kill()
+            }
+        }
+
+        private IEnumerable<Exception> KillTree()
+        {
+            List<Exception> exceptions = new List<Exception>();
```

Comment:
It would be nice if the implementation would treat these cases the same:
Process instance knows the Process has exited
OS returns ESRCH because the process no longer exist
Currently one gives a InvalidOperationException (on HaveNonExitedId) that is swallowed and the other adds a Win32Exception exception to the list.

Reply:
I'd suggest to file issue for that (or perhaps 1 joint issue for all remaining feedback) and address it separately - this PR is already large enough as is.

Reply:
Related: https://github.com/dotnet/corefx/issues/35257 -- if implemented, would resolve this issue.